core/src/main/java/org/stellar/anchor/client/ClientFinder.java

```diff
@@ -42,7 +43,7 @@ public String getClientName(String clientDomain, String account)
       throw new SepNotAuthorizedException("Client not found");
     }
 
-    if (clientDomain != null && !isDomainAllowed(client.getDomain())) {
+    if (clientDomain != null && !isClientDomainsAllowed(client)) {
```

Comment:
You are right, but only if we know the `client` is in `clientAllowList` for sure. I saw the only call site is the Sep10Service, where the account and domain were extracted from request and passed down here, but does it guarantee the client find is allowed?

If this is the case we should also either move or rename the function indicating this is for Sep10 use only, `ClientFinder` is just too general and the validation should be there for other cases.

Reply:
Right because this function is trying to check if the `client` (found through `account` and `domain` extracted from the sep10 request) is allowed, if not then the sep10 request should be reject.